replay/ann/ann_mixin.py

```diff
@@ -0,0 +1,195 @@
+import uuid
+from abc import abstractmethod
+from functools import cached_property
+from typing import Optional, Dict, Union, Any
+
+from pyspark.sql import DataFrame
+from pyspark.sql import functions as sf
+
+from replay.models.base_rec import BaseRecommender
+
+
+class ANNMixin(BaseRecommender):
```

Comment:
please add dockstrings for each method and add short description of adding new model with ann inference to documentation (docs/pages/installation.rst -> Adding new model)